fmskill/model.py

```diff
@@ -92,9 +97,42 @@ def _validate_observation(self, observation) -> bool:
         elif self.is_dfs0:
             # TODO: add check on name
             ok = True
+        return ok
+
+    def _validate_item_eum(self, observation, mod_item):
+        ok = True
+        try:
+            mod_item = self._parse_model_item(mod_item)
+        except Exception:
+            warnings.warn(
+                "Could not parse model item eum type. Consider updating mikeio to version 0.6.4."
+            )
+            return ok
+
+        obs_item = observation.itemInfo
+        if obs_item.type == eum.EUMType.Undefined:
+            warnings.warn(f"{observation.name}: Cannot validate as type is Undefined.")
 
+        if mod_item.type != obs_item.type:
+            ok = False
+            warnings.warn(
+                f"{observation.name}: Item type should match. Model item: {mod_item.type.display_name}, obs item: {obs_item.type.display_name}"
+            )
+        if mod_item.unit != obs_item.unit:
+            ok = False
+            warnings.warn(
+                f"{observation.name}:Unit should match. Model unit: {mod_item.unit.display_name}, obs unit: {obs_item.unit.display_name}"
+            )
         return ok
 
+    def _parse_model_item(self, mod_item):
+        # this will fail for dfs0 in mikeio <0.6.4
+        dfsItemInfo = self.dfs._source.ItemInfo
+
+        mod_item_num = _valid_item_numbers(dfsItemInfo, mod_item)
```

Comment:
How about using the items property instead?https://github.com/DHI/mikeio/blob/dba19843234f9d82f16f16e9e3afd40487a3c04a/mikeio/dfs0.py#L413

Reply:
I don't know if the user gave the item as a str or an int. That's why I opted for the internal mikeio methods in dfsutil.py. Could you please sketch how I could achieve this in a better way?

Reply:
https://github.com/DHI/fmskill/pull/40/commits/7fb38a3099f096fab8d8a6409421b05f16107eba